aten/src/ATen/native/cudnn/Conv_v8.cpp

```diff
@@ -162,27 +161,27 @@ struct BenchmarkCache {
 std::mutex mutex;
 std::unordered_map<KeyType, cudnn_frontend::ExecutionPlan, ParamsHash<KeyType>, ParamsEqual<KeyType>> engine_cache;
 
-// TODO: is this thread safe if cache is updated? is pointer stale?
+// no mutexes here as caches are now thread local for v8, can also return a pointer
+// to the Execution Plan if we know it will not be invalidated by another thread
 cudnn_frontend::ExecutionPlan* find(const KeyType& key) {
-  std::lock_guard<std::mutex> guard(mutex);
   auto it = engine_cache.find(key);
   if (it == engine_cache.end()) {
     return nullptr;
   }
-  // TODO: probably want ExecutionPlan copy constructor or better way to return
   return &(it->second);
 }
 
 void update(const KeyType& key, T& results) {
-  std::lock_guard<std::mutex> guard(mutex);
   engine_cache.erase(key);
```

Comment:
Would this auto release `cudnn_frontend::ExecutionPlan`?

Reply:
Do you mean release in terms of thread safety, or something else (memory management)?

In this case by making the caches `thread_local` each thread would have its own execution plans, even if the workload is the same.

If you mean memory management, then I believe it should call the destructor and then it's cuDNN's responsibility to ensure that the `ExecutionPlan` is properly deallocated. In general, we would expect `erase` on a key that is already in the cache to be rare---it should only happen when an OOM due to a workspace is hit.